### PR TITLE
Update Microsoft.VisualStudio.Sdk.TestFramework to 17.2.7

### LINKED ISSUE #11977: Update Microsoft.VisualStudio.Sdk.TestFramework to 17.2.7
The `Microsoft.VisualStudio.Sdk.TestFramework` should be updated to the Latest Stable, which is currently `17.2.7`.
This task is a step of https://github.com/NuGet/Client.Engineering/issues/438.

Note: https://github.com/NuGet/Home/issues/11975 was found when performing this upgrade (and is already fixed)

### PR DESCRIPTION
**Pull request by donnie-msft**
## Bug

Fixes: https://github.com/NuGet/Home/issues/11977

Regression? Last working version:

## Description

Updating to latest stable of Microsoft.VisualStudio.Sdk.TestFramework which is `17.2.7`. This also requires a Moq upgrade.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/NuGet.PackageManagement.VisualStudio.Test.csproj [hunk 1]
```diff
@@ -89,6 +89,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.VisualStudio.Sdk.TestFramework" />
+    <PackageReference Include="Microsoft.VisualStudio.Sdk.TestFramework.Xunit" />
```
Comment (nkolev92) [inline test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/NuGet.PackageManagement.VisualStudio.Test.csproj:91]: This package depends on [Microsoft.VisualStudio.Sdk.TestFramework](https://www.nuget.org/packages/Microsoft.VisualStudio.Sdk.TestFramework/)

Do we need to have both references? Whenever you install a test framework specific package usually the idea is that you don't need the generic one.

**Review by nkolev92 — DISMISSED** [review]
Quick question about reducing the number of references.
Comfortable merging as is if you prefer it.

File: test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/NuGet.PackageManagement.VisualStudio.Test.csproj (hunk 1 above)
Comment (donnie-msft) [reply test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/NuGet.PackageManagement.VisualStudio.Test.csproj:91]: Good point, I think I can remove them. Running a build now..

File: build/packages.targets [hunk 2]
```diff
@@ -101,8 +101,9 @@
         <PackageReference Update="Microsoft.TestPlatform.Portable" Version="17.1.0" />
         <PackageReference Update="Microsoft.PowerShell.3.ReferenceAssemblies" Version="1.0.0" />
         <PackageReference Update="Microsoft.Test.Apex.VisualStudio" Version="17.1.0-preview-2-31918-026" />
-        <PackageReference Update="Microsoft.VisualStudio.Sdk.TestFramework" Version="17.0.4-alpha" />
-        <PackageReference Update="Moq" Version="4.16.1" />
+        <PackageReference Update="Microsoft.VisualStudio.Sdk.TestFramework" Version="17.2.7" />
```
Comment (nkolev92) [inline build/packages.targets]: you probably don't need this anymore (same as before, up to you).

**Review by nkolev92 — DISMISSED** [review]

**Review by zivkan — APPROVED** [review]

**Pull request merged**